ppl/src/main/java/com/amazon/opendistroforelasticsearch/sql/ppl/parser/AstBuilder.java

```diff
@@ -0,0 +1,194 @@
+import com.amazon.opendistroforelasticsearch.sql.ppl.plans.logical.LogicalPlan;
+import com.amazon.opendistroforelasticsearch.sql.ppl.plans.logical.Project;
+import com.amazon.opendistroforelasticsearch.sql.ppl.plans.logical.Relation;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.antlr.v4.runtime.tree.ParseTree;
+
+import static com.amazon.opendistroforelasticsearch.sql.ppl.antlr.parser.OpenDistroPPLParser.DedupCommandContext;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.antlr.parser.OpenDistroPPLParser.EvalCommandContext;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.antlr.parser.OpenDistroPPLParser.FieldsCommandContext;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.antlr.parser.OpenDistroPPLParser.FromClauseContext;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.antlr.parser.OpenDistroPPLParser.PplStatementContext;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.antlr.parser.OpenDistroPPLParser.RenameCommandContext;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.antlr.parser.OpenDistroPPLParser.SearchFilterFromContext;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.antlr.parser.OpenDistroPPLParser.SearchFromContext;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.antlr.parser.OpenDistroPPLParser.SearchFromFilterContext;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.antlr.parser.OpenDistroPPLParser.SortCommandContext;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.antlr.parser.OpenDistroPPLParser.StatsCommandContext;
+import static com.amazon.opendistroforelasticsearch.sql.ppl.antlr.parser.OpenDistroPPLParser.WhereCommandContext;
+
+/**
+ * Class of walking the AST
+ * Refines the visit path and build the LogicalPlan and Expression nodes interface
+ */
+@RequiredArgsConstructor
+public class AstBuilder extends OpenDistroPPLParserBaseVisitor<LogicalPlan> {
+    private final AstExpressionBuilder expressionBuilder;
+
+    @Override
+    public LogicalPlan visitPplStatement(PplStatementContext ctx) {
+        LogicalPlan search = visit(ctx.searchCommand());
+        LogicalPlan reduce = ctx.commands().stream().map(this::visit).reduce(search, (r, e) -> e.withInput(r));
+        return reduce;
+    }
+
+    /** Search command */
+    @Override
+    public LogicalPlan visitSearchFrom(SearchFromContext ctx) {
+        return visitFromClause(ctx.fromClause());
+    }
+
+    @Override
+    public LogicalPlan visitSearchFromFilter(SearchFromFilterContext ctx) {
+        return new Filter(visitExpression(ctx.logicalExpression())).withInput(visit(ctx.fromClause()));
+    }
+
+    @Override
+    public LogicalPlan visitSearchFilterFrom(SearchFilterFromContext ctx) {
+        return new Filter(visitExpression(ctx.logicalExpression())).withInput(visit(ctx.fromClause()));
+    }
```

Comment:
Why we want to different SearchFromFilter and SearchFilterFrom?

Reply:
It was only the matter of the order in grammar file